### DISCUSSION TITLE
How to add Authorization: Bearer token

### DISCUSSION THREAD

**Discussion by valdemarpavesi** [grafana/grafana-json-datasource#55] [Q&A] [answered]
Hello Marcus,

thanks for your work and nice solution!

Will this plugin be part of the Grafana plugin base?

Another question: How to add  Authorization: Bearer token?

regards!
Valdemar

**Comment by marcusolsson** [comment, marked as answer]
Hi Valdemar, and thank you!

There's no current plans to add this to Grafana itself. Maybe in the future, but right now it's easier to fix and publish fixes faster since it's not tied to the Grafana release cycle.

There's no built-in authentication for something like OAuth right now. You'd have to manually add it as a custom header in the data source config. For more advanced authentication, please submit a feature request.

**Reply by valdemarpavesi** [reply]
thanks @marcusolsson

**Reply by gpinkos** [reply]
@marcusolsson Can you explain how to "manually add it as a custom header in the data source config"?  Here is my Settings page:

![image](https://user-images.githubusercontent.com/97072938/172272455-ddb600d1-c009-491d-a485-456821ea1c48.png)

and here is the documentation from the API which I am trying to use: https://tsheetsteam.github.io/api_docs/?shell#retrieve-timesheets

I tested my URL and bearer token in Node-RED using HTTP GET request and was able to get the data fine, so I know my URL and token work.  It's only in this plugin that I cannot figure out how to insert my token as a Custom HTTP Header

PS:  I also tried with the OAuth toggle selected (instead of With Credentials) and got the same result.

**Reply by gpinkos** [reply]
I have solved my question with the help of someone on the Grafana forum.

Move `Bearer` to Value

Header = `Authorization`
Value = `Bearer SomeAuthKey`
